northstar/src/api/client.rs

```diff
@@ -156,78 +155,16 @@ impl<'a> Client {
         notifications: Option<usize>,
         timeout: time::Duration,
     ) -> Result<Client, Error> {
-        let (notification_tx, notification_rx) = mpsc::channel(1000);
-        let (request_tx, mut request_rx) =
-            mpsc::channel::<(ClientRequest, oneshot::Sender<Result<Response, Error>>)>(10);
-        let mut response_tx = Option::<oneshot::Sender<Result<Response, Error>>>::None;
-
-        let mut connection = time::timeout(timeout, Self::connect(url, notifications, timeout))
+        let connection = time::timeout(timeout, Self::connect(url, notifications, timeout))
             .await
             .map_err(|_| Error::Timeout)??;
 
         debug!("Connected to {}", url);
 
-        task::spawn(async move {
-            loop {
```

Comment:
removing the task indeed simplifies the code...nice

Reply:
but how do notifications work when no request is pending?

Reply:
The `Stream` impl of `Client` checks the  notification buffer and if that's empty it polls the connection.